### PR TITLE
fix(core): rating indicator, fix a11y issues

### PR DESCRIPTION
**Pull request by Betrozov**
## Related Issue.
Closes https://github.com/SAP/fundamental-ngx/issues/6183

## Description
Fixed rating indicator a11y issues

##### During Implementation
1. Visual Testing:
- [x] visual misalignments/updates
- [x] check Light/Dark/HCB/HCW themes
- [n/a] RTL/LTR - proper rendering and labeling
- [x] responsiveness(resize)
- [x] Content Density (Cozy/Compact/(Condensed))
- [x] States - hover/disabled/focused/active/on click/selected/selected hover/press state
- [x] Interaction/Animation - open/close, expand/collapse, add/remove, check/uncheck
- [x] Mouse vs. Keyboard support
- [n/a] Text Truncation
2. API and functional correctness
- [x] check for console logs (warnings, errors)
- [n/a] API boundary values
- [n/a] different combinations of components - free style
- [n/a] change the API values during testing
3. Documentation and Example validations
- [n/a] missing API documentation or it is not understandable
- [x] poor examples
- [x] Stackblitz works for all examples
4. Accessibility testing
5. Browser Testing - Edge, Safari, Chrome, Firefox

##### PR Quality
- [x] the commit message(s) follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/main/CONTRIBUTING.md
- [n/a] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist
- [x] Run npm run build-pack-library and test in external application
- [n/a] update `README.md`
- [n/a] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)

### CODE REVIEW

File: libs/core/src/lib/rating-indicator/pipes/rating-star-label.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Pipe, PipeTransform } from '@angular/core';
+
+@Pipe({name: 'ratingStarLabel'})
+export class RatingStarLabelPipe implements PipeTransform {
+    transform(index: number, controlsCount: number, useHalves: boolean): string {
+        if (useHalves) {
+           return `${index / 2 + 0.5} of ${controlsCount / 2}`
```
Comment (platon-rov) [inline libs/core/src/lib/rating-indicator/pipes/rating-star-label.pipe.ts:7]: Let's use `;` at the end of line :)

**Review by platon-rov — APPROVED** [review]

**Review by Smiranin — APPROVED** [review]

**Comment by dmitry-stepanenko** [comment]
@puru-hk browser allows to navigate through options using "shift + left/right arrow". Additionally, VoiceOver supports changing value by "ctrl+option+left/right arrow". Isn't it enough? 
If needed, I can add explicit support for "left/right arrow" without any other buttons. Should I?

**Comment by puru-hk** [comment]
@dmitry-stepanenko Please add the keyboard support to access the ‘Rating Indicators’. Because all these icons is accessible through the mouse. This one of accessibility requirement.

**Comment by puru-hk** [comment]
image

OB: There is no visible focus provided for the Rating Indicator.
EB: Ideally, visible focus should be provided to the user to identify where the UI elements is available.

**Comment by puru-hk** [comment]
@Betrozov 

image

OB: The rating indicator is accessed, when any of the rating indicator is selected within the list and if any of the rating is not marked, such rating indicator don't have keyboard support.
EB: Ideally, the keyboard support should be provided to rating indicator if no indicator is selected.

**Comment by dmitry-stepanenko** [comment]
@puru-hk I can see the visible outline, when rating indicator is accessed using tab. And it's not required to click on it first to begin to interact. 

The only thing here is there's no visible outline, when indicator has `0` value, but I'm not sure what should be the expected behavior here
Oct-06-2021 15-44-59

**Comment by Sanjay-1312** [comment]
Hello Colleagues,
image

The value is set to '0' and its default values, the values cannot be changed in the left side and the right side of the rating indicator is not reachable using the keyboard.
The rating indicator can be selected using mouse the same should be reachable using keyboard.
So please fix the issue

**Comment by Sanjay-1312** [comment]
Hello Colleagues,

As discussed with Purushotham and Manjunath in the call.
The Rating indicators are the examples of the different sizes of the stars used.
Hence its okay we don't have keyboard support with '0' value Rating indicator with no visible focus for these type of examples.

Hence this issue can be closed.

Thanks and Regards
Sanjay

**Review by puru-hk — APPROVED** [review]

**Pull request merged**